### PR TITLE
EventGrid: Regenerate new SDK for 2019-01-01 Api Version

### PR DESCRIPTION
**Pull request by ahamad-MS**
## Description

Regenerate new SDK for API Version 2019-01-01 based off the following new Swagger Spec: https://github.com/Azure/azure-rest-api-specs/pull/4899 
---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [X] Please add REST spec PR link to the SDK PR
https://github.com/Azure/azure-rest-api-specs/pull/4899

- [X] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [X] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [X] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [X] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [X] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
https://github.com/Azure/azure-rest-api-specs/pull/4899

- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/EventGrid/DataPlane/Microsoft.Azure.EventGrid.Tests/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -17,6 +17,6 @@
 
 [assembly: ComVisible(false)]
 
-[assembly: AssemblyVersion("1.0.0.0")]
-[assembly: AssemblyFileVersion("1.0.0.0")]
+[assembly: AssemblyVersion("2.0.0.0")]
```
Comment (kalyanaj) [inline src/SDKs/EventGrid/DataPlane/Microsoft.Azure.EventGrid.Tests/Properties/AssemblyInfo.cs]: The change in this file should be reverted - this is the dataplane AssemblyInfo.

File: src/SDKs/EventGrid/DataPlane/Microsoft.Azure.EventGrid/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -8,8 +8,8 @@
 [assembly: AssemblyTitle("Microsoft Azure EventGrid Library")]
 [assembly: AssemblyDescription("Provides developers with a library to publish events to Azure EventGrid and to consume events delivered by EventGrid.")]
 
-[assembly: AssemblyVersion("3.0.0.0")]
-[assembly: AssemblyFileVersion("3.0.0.0")]
+[assembly: AssemblyVersion("4.0.0.0")]
```
Comment (kalyanaj) [inline src/SDKs/EventGrid/DataPlane/Microsoft.Azure.EventGrid/Properties/AssemblyInfo.cs]: The change in this file should be reverted as this is the dataplane assemblyinfo.

Comment (ahamad-MS) [reply src/SDKs/EventGrid/DataPlane/Microsoft.Azure.EventGrid/Properties/AssemblyInfo.cs]: fix

File: src/SDKs/EventGrid/DataPlane/Microsoft.Azure.EventGrid.Tests/Properties/AssemblyInfo.cs (hunk 1 above)
Comment (ahamad-MS) [reply src/SDKs/EventGrid/DataPlane/Microsoft.Azure.EventGrid.Tests/Properties/AssemblyInfo.cs]: fixed

File: src/SDKs/EventGrid/management/Management.EventGrid/Microsoft.Azure.Management.EventGrid.csproj [hunk 3]
```diff
@@ -6,10 +6,10 @@
   <PropertyGroup>
     <PackageId>Microsoft.Azure.Management.EventGrid</PackageId>
     <Description>Provides developers with a library to create and manage all Azure EventGrid resources. </Description>
-    <Version>3.0.1-preview</Version>
+    <Version>4.0.0</Version>
     <AssemblyName>Microsoft.Azure.Management.EventGrid</AssemblyName>
     <PackageTags>Microsoft Azure EventGrid Management;Event Grid;Event Grid management;</PackageTags>
-    <PackageReleaseNotes>Taking dependency on 10.0.3 version of Newtonsoft nuget package.</PackageReleaseNotes>
+    <PackageReleaseNotes>Introducing 2019-01-01 as new GA API version.</PackageReleaseNotes>
```
Comment (kalyanaj) [inline src/SDKs/EventGrid/management/Management.EventGrid/Microsoft.Azure.Management.EventGrid.csproj]: Suggest replacing this with "This corresponds to the 2019-01-01 stable API version that supports General Availability of the following functionality related to event subscriptions: DeadLetter destination, Storage queue as destination, HybridConnection as destination, manual handshake validation, and support for retry policies. Features that are still in preview (such as Event Grid domains or advanced filters support) can still be accessed using the 3.0.1-preview version of the SDK."

**Comment by ahamad-MS** [comment]
@dsgouda .. Can you please take a look at this PR or suggest someone? thx

File: src/SDKs/EventGrid/management/Management.EventGrid/Microsoft.Azure.Management.EventGrid.csproj (hunk 3 above)
Comment (ahamad-MS) [reply src/SDKs/EventGrid/management/Management.EventGrid/Microsoft.Azure.Management.EventGrid.csproj]: fixed

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
Please rebase and clean up your commits. Keep logical commits in the history.
Also use generate.ps1 to generate your SDK. You are missing meta data that get's generated as part of the SDK generation process.

**Comment by ahamad-MS** [comment]
Restarting new PR with fewer commits

**Pull request closed without merging**